File: app/src/main/java/com/seafile/seadroid2/framework/model/sdoc/FileLinkModel.java
package com.seafile.seadroid2.framework.model.sdoc;

import android.os.Parcel;
import android.os.Parcelable;

public class FileLinkModel implements Parcelable {
    public String display_value;
    public String row_id;

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.display_value);
        dest.writeString(this.row_id);
    }

    public FileLinkModel() {
    }

    protected FileLinkModel(Parcel in) {
        this.display_value = in.readString();
        this.row_id = in.readString();
    }

    public static final Parcelable.Creator<FileLinkModel> CREATOR = new Creator<FileLinkModel>() {
        @Override
        public FileLinkModel createFromParcel(Parcel source) {
            return new FileLinkModel(source);
        }

        @Override
        public FileLinkModel[] newArray(int size) {
            return new FileLinkModel[size];
        }
    };
}

File: app/src/main/java/com/seafile/seadroid2/framework/model/sdoc/FileLinkModel.kt
package com.seafile.seadroid2.framework.model.sdoc

import android.os.Parcel
import android.os.Parcelable

class FileLinkModel() : Parcelable {
    @JvmField
    var display_value: String? = null

    @JvmField
    var row_id: String? = null

    override fun describeContents(): Int = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(display_value)
        dest.writeString(row_id)
    }

    private constructor(parcel: Parcel) : this() {
        display_value = parcel.readString()
        row_id = parcel.readString()
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<FileLinkModel> = object : Parcelable.Creator<FileLinkModel> {
            override fun createFromParcel(source: Parcel): FileLinkModel = FileLinkModel(source)
            override fun newArray(size: Int): Array<FileLinkModel?> = arrayOfNulls(size)
        }
    }
}
